### ISSUE TITLE
Testcase Syntax

### ISSUE THREAD

**Issue by TeamSPoon** [trueagi-io/metta-morph#19]
Going to suggest for testcases (since this works in Rust MeTTa) we adopt a standard test writing format

```
; it is non-deterministic: each value is matched against all cases
!(assertEqual
 (case (+ 1 (superpose (1 2 3)))
  ((3 OK-3)
   (4 OK-4)))
 (superpose (OK-3 OK-4)))
```

```
; one case can produce multiple results
!(assertEqual
  (case (+ 1 (superpose (1 2 3)))
     (($x (+ 1 $x))))
  (superpose (3 4 5)))
  ```

**Comment by TeamSPoon** [comment]
Most tests seem to be described with the addition of these two functions
  
`assertEqualToResult`
`assertEqual`

```
(isa red color)
(isa green color)
(isa blue color)
!(assertEqualToResult
  (match &self (isa $color color) $color)
  (red green blue))
  ```

**Comment by patham9** [comment]
The tests in MeTTamorph directly run both MeTTa and MeTTamorph on the same test files each and check whether the output matches, this is the reason why such assertions are unnecessary. Producing the same output is the implicit assertion which is reported when it fails.

**Comment by Necr0x0Der** [comment]
Yes, I agree that comparing results from compiled and not compiled versions is a good approach for this project, but it may turn out to be the case that something in addition is needed. Some MeTTa code cannot be supported by the compiler and running compiled functions from such the code should also be checked. Thus, I suppose that more tests will be added, which will not be completely compiled. It will still be possible to compare partly-compiled and not compiled versions, but writing such tests may appear not too convenient. Let's try and see how it goes.

**Comment by TeamSPoon** [comment]
Why i was suggesting it is that most all tests in  hyperon-experimental https://github.com/trueagi-io/hyperon-experimental/tree/main/python/tests/scripts  use `assertEqualToResult` `assertEqual`  and didn't see wanting to have to edit those tests files just for metta-morph :)

**Comment by Necr0x0Der** [comment]
If we'd like to run all the unit tests from the main repo with the use of compilation by metta-morph, this would make perfect sense. Well, it also makes sense to be able to run these tests manually to check the results under metta-morph. Thus, I agree that it makes sense to introduce asserts for this sake, although it doesn't imply that unit tests in metta-morph itself should necessarily use them.

**Comment by patham9** [comment]
Good point, we can add these assert functions at least.
